### PR TITLE
Removing xen-info and its need for state

### PR DESCRIPTION
**Pull request by rvs**
This is a final bit of refactoring that allows us to get rid of storing the intermediate result in the r/w part of the filesystem (there's no need for that anymore).

While working on this I noticed that if we get the unknown state from `xl info` we declare domain as `running` a few times. I am not sure whether this is better of worse than declaring it whatever the last state it was in -- but that's the current behavior anyway -- so I'm preserving it.

### CODE REVIEW

**Review by lf-edge-eve — APPROVED** [review]
Let's test this

**Comment by mydatascience** [comment]
Please rerun Eden. Fixed.

File: pkg/xen-tools/xen-start [hunk 1]
```diff
@@ -46,13 +93,12 @@ xl unpause "$ID" || bail "xl unpause failed"
 # now star polling for domain status (11 sec interval) in the background
 # (note our use of mv to make sure file reads on the other side are atomic)
 (while true; do
-   sleep 11
```
Comment (giggsoff) [inline pkg/xen-tools/xen-start:49]: Hmm, seems, that we miss sleep here.

Comment (rvs) [reply pkg/xen-tools/xen-start:49]: miss in what sense @giggsoff ?

Comment (giggsoff) [reply pkg/xen-tools/xen-start:49]: I mean, that now we have the loop with xl info without sleep.

**Pull request merged**